src/main/java/fr/moribus/imageonmap/commands/maptool/DeleteCommand.java

```diff
@@ -65,29 +68,74 @@ private static RawText deleteMsg(Class klass, ImageMap map) {
                 .then(I.t("[Confirm]"))
                 .color(ChatColor.GREEN)
                 .hover(new RawText(I.t("{red}This map will be deleted {bold}forever{red}!")))
-                .command(klass, map.getId(), "--confirm")
+                .command(klass, sender.getName(), map.getId(), "--confirm")
                 .build();
     }
 
     @Override
     protected void run() throws CommandException {
-        ImageMap map = getMapFromArgs();
+        ArrayList<String> arguments = getArgs();
+        final boolean confirm = hasFlag("confirm");
 
-        if (!hasFlag("confirm")) {
-            RawText msg = deleteMsg(getClass(), map);
-            send(msg);
-        } else {
-            Player player = playerSender();
-            MapManager.clear(player.getInventory(), map);
-
-            try {
-                MapManager.deleteMap(map);
-                info(I.t("Map successfully deleted."));
-            } catch (MapManagerException ex) {
-                PluginLogger.warning(I.t("A non-existent map was requested to be deleted", ex));
-                warning(I.t("This map does not exist."));
+        if (arguments.size() > 3 || (arguments.size() > 2 && !confirm)) {
+            warning(I.t("Too many parameters! Usage: /maptool delete [playername]:<mapname>"));
```

Comment:
Aussi pas besoin de préciser l'usage, juste le message d'erreur. QuartzLib rappelle l'usage dans le message d'argument invalide. Il faut cela dit le mettre à jour dans `@CommandInfo`. Je me peux ajouter de suggestion en ce sens car c'est hors du diff, GitHub ne m'offre pas l'option.

```suggestion
            throwInvalidArgument(I.t("Too many parameters!"));
```

Reply:
Bon à savoir, j'ajoute ça sur le prochain comit